### PR TITLE
ui changes

### PR DESCRIPTION
**Pull request by pablo-clueless**
new: dashboard
new: purchase workflow
new: profile and password edit
mod: auth workflow

### CODE REVIEW

File: src/components/instant-buy/success.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import React from "react"
+
+import { Button } from ".."
+
+interface Props {
+	txnHash: string
+}
+
+const Success = (props: Props) => {
+	const viewTxn = () => {
```
Comment (Extheoisah) [inline src/components/instant-buy/success.tsx:12]: We don't need this function. wrap it in a link that opens a new tab. 
No need to call window

File: src/components/profile/authentication.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"use client"
+const Authentication = () => {
+	return (
+		<div className="h-[644px] w-full rounded-lg border border-black-500 bg-black-700 p-10">
+			<div className="flex w-full items-center justify-between">
+				<div className="flex items-center gap-5">
+					<div className="aspect-square w-[120px] rounded-full bg-alt-orange-100"></div>
+					<div>
+						<p className="font-satoshi text-2xl font-bold">Pablo Clueless</p>
+						<p className="text-white-300">smsnmicheal@gamil.com</p>
```
Comment (Extheoisah) [inline src/components/profile/authentication.tsx:10]: fetch profile details from the backend

File: src/components/sidebar.tsx
Comment (Extheoisah) [inline src/components/sidebar.tsx:1]: lets delete this file if we no longer need it

**Review by Extheoisah — CHANGES_REQUESTED** [review]
Thanks for opening this for early review. There are a lot of changes related to changes in prettier configuration (the tab spaces most especially). I would suggest we do not change that.

File: .npmrc
[the remark this answers is not in the document]
Comment (pablo-clueless) [reply .npmrc:1]: Defines how npm should behave whenever a command is run. For now, I'm just making sure that dependencies of every package is also installed along with it.

File: src/app/account/forgot-password/page.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+							value={email}
+							onChange={(e) => setEmail(e.target.value)}
+							label="Email Address"
+						/>
+					</div>
+					<div className="mt-[450px] flex w-full flex-col gap-5">
+						<div className="grid w-full grid-cols-2 gap-6">
+							<Button
+								type="button"
+								onClick={() => router.back()}
```
[the remark this answers is not in the document]
Comment (pablo-clueless) [reply src/app/account/forgot-password/page.tsx:82]: We're sure of the previous page would always be `login`, that's the only place where this page can be accessed.

File: src/app/account/forgot-password/page.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+		<>
+			{error && (
+				<Dialog
+					isOpen={error !== ""}
+					onDismiss={() => setError("")}
+					title="Reset Error!"
+					type="error"
+					large
+					description="An error occured. Please try again.">
+					<div></div>
```
[the remark this answers is not in the document]
Comment (pablo-clueless) [reply src/app/account/forgot-password/page.tsx:49]: The `children` props was not optional, forgot to switch it.

File: src/app/account/forgot-password/page.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+			)}
+			{success && (
+				<Dialog
+					isOpen={success}
+					onDismiss={() => setSuccess(false)}
+					title="Success!"
+					type="success"
+					large
+					description="Instructions to reset your pasword has been sent to your mail.">
+					<div></div>
```
[the remark this answers is not in the document]
Comment (pablo-clueless) [reply src/app/account/forgot-password/page.tsx:60]: The `children` props was not optional, forgot to switch it.

File: src/app/api/transactions/route.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+				)
+			}
+		} catch (error) {
+			return NextResponse.json(
+				{ success: false, message: "Internal server error" },
+				{ status: 500 }
+			)
+		}
+	}
+	try {
```
[the remark this answers is not in the document]
Comment (pablo-clueless) [reply src/app/api/transactions/route.ts:45]: Two requests. If the URL contains a `txn` query string, we check for a transaction by it's ID, else we return all transactions.

File: src/app/dashboard/nav-list.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+export const NavList = [
+	{
+		icon: <SquaresFour size={24} />,
+		label: "dashboard",
+		path: "/dashboard",
+	},
+	{
+		icon: <CalendarBlank size={24} />,
+		label: "plans",
+		path: "/dashboard/plans",
```
[the remark this answers is not in the document]
Comment (pablo-clueless) [reply src/app/dashboard/nav-list.tsx:20]: They all have the same layouts and are only accessible from the dashboard. The other alternative is to create a layout, and reuse wherever needed. It adds more files/folders.

File: src/app/dashboard/page.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+								}>
+								{CurrencyList.map((currency) => (
+									<option key={currency} value={currency}>
+										{currency}
+									</option>
+								))}
+							</CurrencyInput>
+							<p className="flex items-center gap-1 text-xs text-black-400">
+								<WarningCircle className="text-alt-orange-100" />
+								Exchange rate: 1BTC ={" "}
```
[the remark this answers is not in the document]
Comment (pablo-clueless) [reply src/app/dashboard/page.tsx:80]: The function throws an error for something called requestAsyncStorage

File: src/components/instant-buy/init.tsx [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+						{CurrencyList.map((currency) => (
+							<option key={currency} value={currency}>
+								{currency}
+							</option>
+						))}
+					</CurrencyInput>
+				</div>
+				<p className="flex items-center gap-1 text-xs text-black-400">
+					<WarningCircle className="text-alt-orange-100" />
+					Exchange rate: 1BTC ={" "}
```
[the remark this answers is not in the document]
Comment (pablo-clueless) [reply src/components/instant-buy/init.tsx:107]: The function throws an error about something called requestAsyncStorage

File: src/components/instant-buy/init.tsx [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+	const handleSubmit = () => {
+		const { amount, walletAddress } = fields
+		if (!amount) {
+			return alert("Please enter amount!")
+		}
+		if (!walletAddress) {
+			return alert("Please enter wallet address!")
+		}
+		props.setDepositInfo({
+			accountNumber: "988852055355",
```
[the remark this answers is not in the document]
Comment (pablo-clueless) [reply src/components/instant-buy/init.tsx:47]: The endpoint for getting payment details is throwing errors both here and in Postman.

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**